src/auth.js

```diff
@@ -34,7 +34,14 @@ async function retrieveToken(method, client) {
             const githubAudience = core.getInput('jwtGithubAudience', { required: false });
 
             if (!privateKey) {
-                jwt = await core.getIDToken(githubAudience)
+                try {
+                    core.debug("Fetching ID token from GitHub OIDC provider")
+                    jwt = await core.getIDToken(githubAudience)
+                } catch (err) {
+                    core.error("Error fetching ID token from GitHub OIDC provider")
```

Comment:
I'm not sure if it'd be better to be more explicit here with the messaging. For example, should I change this to "Error calling core.getIDToken"?

Reply:
I guess I'm wondering what the goal of catching this error is? Does it happen currently or is being hidden? If we aren't doing something with the error and just re-throwing it, will it be hidden again?

Reply:
I'm not sure if it's currently happening or being hidden. I've seen some `triggerUncaughtException(err, true /* fromPromise */)` errors in the logs and that's caused by not catching errors thrown by promises.
I believe we have to rethrow it in order for the workflow to fail.

I'm basically copying [this change](https://github.com/hashicorp/vault-action/commit/8708b58476ed098be5f1169c405233c2850ec893) Allan added to get more error output, but I'm going a level down.

This is also adding an error annotation, so I'm not sure if we want that or if we want `core.setOutput`.

Reply:
>  errors in the logs and that's caused by not catching errors thrown by promises.
I believe we have to rethrow it in order for the workflow to fail.

Is there any way we test this manually or in the code to make sure? I'm really not familiar with JS at all so I don't have a feel for what is really happening. If you know this will uncover those errors then that is fine too.

> This is also adding an error annotation, so I'm not sure if we want that or if we want core.setOutput

I think the reason to use `core.setOutput("errormessage", error?.message);` is that we can then inspect the errormessage output outside of the action step. For example https://github.com/github/internal-actions/blob/main/.github/workflows/default.yml#L247

In this instance, I'm not sure `core.setOutput(err?.message)` is correct because it is missing a variable to bind the output to, so I think we might as well make that `core.setOutput("errormessage", err?.message);`?

Reply:
Also wanted to say, we might want to adjust the post processing code to capture and send metrics for the error you are generating, otherwise we will not be able to easily see if these errors are occurring unless we do some manual inspection.

Reply:
> Is there any way we test this manually or in the code to make sure? I'm really not familiar with JS at all so I don't have a feel for what is really happening. If you know this will uncover those errors then that is fine too.
>
I'm trying to find node docs on this, but I got this conclusion from https://stackoverflow.com/questions/71077701/react-jest-nodeinternal-process-promises246-triggeruncaughtexceptionerr-tr. `core.getIDToken` can throw errors so I think we should be catching these anyways. https://github.com/actions/toolkit/blob/c5c786523e095ca3fabfc4d345e16242da34e108/packages/core/src/oidc-utils.ts

Maybe, in this case, we should catch this error, and append our own error message before we rethrow (instead of just throwing the original message)?

> I think the reason to use core.setOutput("errormessage", error?.message); is that we can then inspect the errormessage output outside of the action step. For example https://github.com/github/internal-actions/blob/main/.github/workflows/default.yml#L247
>
> In this instance, I'm not sure core.setOutput(err?.message) is correct because it is missing a variable to bind the output to, so I think we might as well make that core.setOutput("errormessage", err?.message);?
>
I think I was mixing up `core.setOutput` and `console.log` 😆 My js is rusty

Reply:
Yeah, since [`entry.js` will set the `errormessage` output](https://github.com/wagnerm/vault-action/blob/allan-pires/throw-error-at-exportsecrets-step/src/entry.js), I think it might be better to throw the new error (and have it contain the old error). I will make that change 👍 

I'm going to manually test this by either manually throwing an error or by setting one of the inputs to gibberish

Reply:
I think that's better. I'm still not sure where the `triggerUncaughtException(err, true /* fromPromise */)` came from, maybe `getIDToken`?
### Test throwing error with the error handling
<img width="1022" alt="image" src="https://github.com/wagnerm/vault-action/assets/37482914/60e21b45-874a-40b4-83b2-c5594024c34c">
https://github.com/github/internal-actions/actions/runs/6165865722/job/16734398929

### Test throwing error without the error handling
<img width="1015" alt="image" src="https://github.com/wagnerm/vault-action/assets/37482914/03603237-27d2-4136-80cf-faaab15c546c">
https://github.com/github/internal-actions/actions/runs/6165895752/job/16734479869?pr=332

Reply:
I'm not sure if the original error location helps us out that much here, especially in the case of a `triggerUncaughtException(err, true /* fromPromise */` error.

Reply:
Honestly, I can't tell if the original error is getting hidden because my test error was a couple lines above it and I had a bunch of debugs 🤣

Reply:
I'm not sure though if this PR addresses that issue. The original error originated in node, and I don't have the link anymore. Should we just close this?

Reply:
I'm updating this to just log and rethrow the original error